File: Projects/leetcode/check.java
package leetcode;

class LeetCode {
    public static boolean check(int[] nums) {
        boolean isFlipped = false;
        for (int i = 1; i < nums.length; i++) {
            if (nums[i - 1] <= nums[i]) continue;
            else if (isFlipped) return false;
            else isFlipped = true;
        }
        return !isFlipped || isFlipped && nums[0] >= nums[nums.length - 1];
    }
}

File: Projects/leetcode/check.kt
package leetcode

fun check(nums: IntArray): Boolean {
    var isFlipped = false
    for (i in 1 until nums.size) {
        isFlipped = if (nums[i - 1] <= nums[i]) continue else if (isFlipped) return false else true
    }
    return nums[0] >= nums[nums.size - 1]
}
